File: Easy/0171_excel-sheet-column-number/excel-sheet-column-number.cpp
#include <string>

class Solution
{
public:
	int titleToNumber(std::string columnTitle)
	{
		int res = 0;
		for (auto c : columnTitle)
		{
			res *= 26;
			res += (c - 'A' + 1);
		}
		return res;
	}
};

File: Easy/0171_excel-sheet-column-number/excel-sheet-column-number.py
#!/bin/python3

from typing import List

class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        res : int = 0
        for c in columnTitle:
            res = res * 26 + (ord(c) - ord('A') + 1)
        return res

def main():
    tests : List[str] = [
        "A",
        "AB",
        "ZY"
    ]
    for t in tests:
        print(f"---\ntitle: '{t}' number: {Solution().titleToNumber(t)}")

if __name__ == "__main__":
    main()
